front/components/assistant_builder/actions/DustAppRunAction.tsx

```diff
@@ -27,6 +34,7 @@ export function ActionDustAppRun({
   dustApps: AppType[];
 }) {
   const [showDustAppsModal, setShowDustAppsModal] = useState(false);
+  const action = useDeprecatedDefaultSingleAction(builderState);
```

Comment:
Can we refactor to pass the action here instead? This component should be independent of the builderState now?

Reply:
Done for all screens. For now just passing default deprecated action. Next PR, I'll get the right action for each.